src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneObject = {};
+  const arrayWithObjects = [];
+
+  Object.assign(cloneObject, state);
+
+  for (const objects of actions) {
+    const { type, extraData, keysToRemove } = objects;
+
+    switch (type) {
+      case 'addProperties' :
+        Object.assign(cloneObject, extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const deleteValue of keysToRemove) {
+          delete cloneObject[deleteValue];
+        }
+        break;
+
+      case 'clear' :
+        for (const deleteAll in cloneObject) {
```

Comment:
same here, you should think of `what is stored inside the variable` to name it properly